scipy/stats/_multivariate.py

```diff
@@ -5751,9 +5743,66 @@ def _sample_uniform_direction(dim, size, random_state):
     return samples
 
 
+_dirichlet_mn_doc_default_callparams = """\
+alpha : array_like
+    The concentration parameters. The number of entries determines the
+    dimensionality of the distribution.
+n : array_like
+    The number of trials.
+"""
+
+_dirichlet_mn_doc_frozen_callparams = ""
+
+_dirichlet_mn_doc_frozen_callparams_note = """\
+See class definition for a detailed description of parameters."""
+
+dirichlet_mn_docdict_params = {
+    '_dirichlet_mn_doc_default_callparams': _dirichlet_mn_doc_default_callparams,  # noqa
+    '_doc_random_state': _doc_random_state
+}
+
+dirichlet_mn_docdict_noparams = {
+    '_dirichlet_mn_doc_default_callparams': _dirichlet_mn_doc_frozen_callparams,
+    '_doc_random_state': _doc_random_state
+}
+
+def _dirichlet_multinomial_check_parameters(alpha, n, x=None):
+
+    if x is None:
+        alpha = np.asarray(alpha)
+    else:
+        # Ensure that `x` and `alpha` are arrays. If the shapes are
+        # incompatible, NumPy will raise an appropriate error.
+        x, alpha = np.broadcast_arrays(x, alpha)
+
+        x_int = np.floor(x)
+        if np.any(x < 0) or np.any(x != x_int):
+            raise ValueError("`x` must contain only non-negative integers.")
+        x = x_int
+
+    alpha = np.asarray(alpha)
+
+    if np.any(alpha <= 0):
+        raise ValueError("`alpha` must contain only positive values.")
+
+    n_int = np.floor(n)
+    if np.any(n < 0) or np.any(n != n_int):
+        raise ValueError("`n` must be non-negative integer(s).")
+    n = n_int
+
+    sum_alpha = np.sum(alpha, axis=-1)
+
+    return (alpha, sum_alpha, n) if x is None else (alpha, sum_alpha, n, x)
```

Comment:
I collected all the required input validation into this function so it doesn't need to be copy-pasted into every method. I refactored some of them to be written as you might see them elsewhere in `scipy.stats`.